File: java/src/main/java/com/mikeylight/algos/arrays/BubbleSort.java
class BubbleSort {
    static int[] bubbleSort(int[] numbers) {
        if(numbers == null || numbers.length == 1) {
            return numbers;
        }
        boolean isSorted = false;

        while(!isSorted) {
            isSorted = true;
            for(int i = 0; i < numbers.length - 1; i++) {
                if(numbers[i] > numbers[i + 1]) {
                    int hold = numbers[i + 1];
                    numbers[i + 1] = numbers[i];
                    numbers[i] = hold;
                    isSorted = false;
                }
            }
        }
        return numbers;
    }
}

File: python/algos/arrays/bubble_sort.py
def bubble_sort(arr):
    if len(arr) == 0 or len(arr) == 1:
        return arr
    arr_len = len(arr)
    is_sorted = False
    while not is_sorted:
        is_sorted = True
        for x in range(arr_len - 1):
            if arr[x] > arr[x + 1]:
                hold = arr[x + 1]
                arr[x + 1] = arr[x]
                arr[x] = hold
                is_sorted = False
    return arr
